torchdynamo/variables/torch.py

```diff
@@ -147,6 +151,39 @@ def call_function(
             )
         elif not config.dynamic_shapes and self.is_dynamic_shapes(args, kwargs):
             unimplemented(f"dynamic shapes: {self.value.__name__}")
+        elif len(args) > 0 and isinstance(args[0], TensorWithTFOverrideVariable):
+            # This code block implements inlining the __torch_function__
+            # override of a tensor.
+
+            # TODO(future PR): make this implement the full __torch_function__ API
+            # instead of assuming the relevant override is in the first argument.
+            args[0] = args[0].tensor_variable
+
+            # This extracts the original tensor
+            tensor_val = args[0].proxy.node.meta["example_value"]
+
+            # Note: the __func__ attribute call is to ensure the function
+            # does not have `self`
+            # Note: currently this does not guard for the source code of the
+            # override.
+            func_var = UserFunctionVariable(
+                tensor_val.__torch_function__.__func__, **options
+            )
+            type_var = UserDefinedObjectVariable(type(tensor_val), **options)
+
+            # signature:
+            # def __torch_function__(cls, func, types, args=(), kwargs=None):
+            tf_args = (
+                type_var,  # cls
+                self,  # func
+                (type_var,),  # types
+                TupleVariable(args),  # args
+                kwargs,  # kwargs
+            )
+            unwrapped = tx.inline_user_function_return(func_var, tf_args, {})
+            # Wrap the result, to ensure the override survives for the next
+            # function.
+            return TensorWithTFOverrideVariable(unwrapped)
```

Comment:
Does a torch_function always return a wrapped tensor?  Or do we need to check for that?

Reply:
thanks, added the relevant logic